scheduler/reconcile.go

```diff
@@ -796,3 +830,71 @@ func (a *allocReconciler) computeUpdates(group *structs.TaskGroup, untainted all
 
 	return
 }
+func (a *allocReconciler) handleDelayedReschedules(rescheduleLater []*delayedRescheduleInfo, all allocSet, tgName string) allocSet {
+	// Sort by time
+	sort.Slice(rescheduleLater, func(i, j int) bool {
+		return rescheduleLater[i].rescheduleTime.Before(rescheduleLater[j].rescheduleTime)
+	})
+
+	var evals []*structs.Evaluation
+	var allocIDs []string
+	nextReschedTime := rescheduleLater[0].rescheduleTime
+	allocIDToFollowupEvalID := make(map[string]string, len(rescheduleLater))
+	for _, allocReschedInfo := range rescheduleLater {
+		if allocReschedInfo.rescheduleTime.UTC().UnixNano()-nextReschedTime.UTC().UnixNano() < batchedFailedAllocWindowSize.Nanoseconds() {
```

Comment:
`allocReschedInfo.rescheduleTime.Sub(nextReschedTime) < batchedFailedAllocWindowSize`

You don't need to, and shouldn't, be going to the unix nano for all these. Use the time methods